src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1314,4 +1314,13 @@
   <data name="PreferConstCharOverConstUnitStringInStringBuilderTitle" xml:space="preserve">
     <value>Consider using 'StringBuilder.Append(char)' when applicable.</value>
   </data>
+  <data name="DoNotCheckFlagFromDifferentEnumDescription" xml:space="preserve">
+    <value>Calling the 'HasFlag' method on enum of some type to check the flag of another enum type will always return 'false'.</value>
```

Comment:
I tried the below console app and it always throws an unhandled exception:

```csharp
using System;

public class C
{
    [Flags]
    public enum MyEnum { A, B, }
    [Flags]
    public enum OtherEnum { A, }

    static void Main(string[] args)
    {
        MyEnum a = MyEnum.A;
        Console.WriteLine(a.HasFlag(OtherEnum.A));
    }
}
```

```
Unhandled exception. System.ArgumentException: The argument type, 'C+OtherEnum', is not the same as the enum type 'C+MyEnum'.
   at System.Enum.HasFlag(Enum flag)
```

Reply:
We should fix the description.

Reply:
Something like: `'Enum.HasFlag' method expects the 'Enum' argument to be of the same type as the 'Enum' instance on which the method is invoked. If these are different 'Enum' types, an unhandled exception will be thrown at runtime.`

Reply:
Sorry I got confused by another ticket I have created: https://github.com/dotnet/roslyn-analyzers/issues/3528

To sum-up if the enum types on the arg and caller are different the method throws at runtime (whether or not the enums are marked with `Flags`) but if you call the method on a type which isn't marked with `Flags` the method always returns false.